src/nccl_ofi_rdma.c

```diff
@@ -5312,7 +5323,20 @@ static int ep_rail_init(nccl_net_ofi_rdma_ep_t *ep,
 {
 	int ret = 0;
 
-	ret = nccl_ofi_ofiutils_init_connection(FI_VERSION(1, 18), dev_rail->info, dev_rail->domain, &ep_rail->ofi_ep,
+	if (domain_per_endpoint == 1) {
+		ret = fi_domain(dev_rail->fabric, dev_rail->info,
+			&ep_rail->domain, NULL);
+		if (OFI_UNLIKELY(ret != 0)) {
+			NCCL_OFI_WARN("Couldn't open a fabric access domain. RC: %d, ERROR: %s",
+				ret, fi_strerror(-ret));
+			free(ep);
```

Comment:
I don't think `free(ep)` is the right thing to do here, given `ep` is an argument to this function and not allocated here.

Actually, we don't seem to free ep anywhere in the RDMA protocol, which may or may not be a good thing.